File: SmartHotEditCasePlugin/StringPlugin.cs
using System;
using System.Linq;
using System.Collections.Generic;
using SmartHotEditPluginHost;
using SmartHotEditPluginHost.Model;
using System.ComponentModel.Composition;

namespace SmartHotEdit.Plugin
{
    /// <summary>
    /// Description of StringPlugin.
    /// </summary>
    [Export(typeof(SmartHotEditPluginHost.APlugin))]
    public class StringPlugin : APlugin
	{
		
		public StringPlugin(){
    		this.addFunction(new Function("Replace", "Replaces a string in a string", new Function.Transform(this.replace), new List<Argument> { new Argument("oldString", "old string"), new Argument("newString", "new string") }));
		}

		public override String getName(){
			return "String";
		}
		
		public override String getDescription(){
			return "Some functions to modify a string";
		}
		
		String replace(String input, List<Argument> arguments = null){
            if(arguments == null || arguments.Count < 2)
            {
                return null;
            }

            return input.Replace(arguments.ElementAt(0).value, arguments.ElementAt(1).value);
		}
		
	}
}

File: SmartHotEditLuaPlugins/Lua/Plugins/Examples/string_plugin.lua
local pluginhelper = require "pluginhelper"
require "class"
require "baseplugin"

LuaStringPlugin = class(APlugin, function(base)
  APlugin.init(base, 'LuaString', 'Some functions to modify a string')  -- must init base!
  replaceStringFunc = Function('Replace', 'Replaces a string in a string', base.replaceString, {Argument("oldString", "old string"), Argument("newString", "new string")})
   
  base:addFunction(replaceStringFunc)
end)
      
function LuaStringPlugin:replaceString(input, arguments)
	if arguments ~= nil and #arguments == 2 then
		result = string.gsub(input, arguments[0].value, arguments[1].value)
		return result
	else
		print("Nothing replaced, you need 2 arguments")
	end
	return input
end


plugin = LuaStringPlugin()
return plugin